document/document.go

```diff
@@ -27,6 +27,20 @@ type Keyer interface {
 	Key() []byte
 }
 
+// Length returns the length of a document.
+func Length(d Document) (int, error) {
+	if fb, ok := d.(FieldBuffer); ok {
```

Comment:
`FieldBuffer` is usually a pointer, this optimization won't work as-is, we should infer for the pointer instead
```suggestion
	if fb, ok := d.(*FieldBuffer); ok {
```